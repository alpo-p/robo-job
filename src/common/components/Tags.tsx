import { useTheme } from '@react-navigation/native'
import React from 'react'
import { View } from 'react-native'

import { Ionicons } from '@expo/vector-icons'
import RoboText from './RoboText'
import styles from '../styles'
import { TagType } from '../types'

interface Props {
  tags: TagType[]
}

interface TagPropProps {
  tag: TagType
  editStylesForSearchScreen?: boolean
}

export const Tag: React.FC<TagPropProps> = ({
  tag,
  editStylesForSearchScreen,
}) => {
  const { colors } = useTheme()
  const b = editStylesForSearchScreen

  return (
    <View
      style={{
        flexDirection: 'row',
        alignContent: 'center',
        borderWidth: 0.5,
        borderColor: colors.primary,
        borderRadius: 4,
        padding: 4,
        marginRight: 6,
        marginTop: b ? undefined : 8,
        height: styles.commonSize,
      }}
    >
      {editStylesForSearchScreen && (
        <Ionicons name="remove-circle" size={14} color={colors.primary} />
      )}
      <RoboText color={colors.primary} uppercase weight="light" size="tiny">
        {!editStylesForSearchScreen && '#'} {tag.text}
      </RoboText>
    </View>
  )
}

const Tags: React.FC<Props> = ({ tags }) => (
  <View
    style={{
      flexDirection: 'row',
      flexWrap: 'wrap',
    }}
  >
    {tags.map(tag => (
      <Tag tag={tag} key={tag.id} />
    ))}
  </View>
)

export default Tags
