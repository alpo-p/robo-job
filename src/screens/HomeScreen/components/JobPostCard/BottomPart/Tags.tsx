import { useTheme } from '@react-navigation/native'
import React from 'react'
import { View } from 'react-native'

import RoboText from '../../../../../common/components/RoboText'
import styles from '../../../../../common/styles'

interface Props {
  tags: string[]
}

const Tag: React.FC<{ tag: string }> = ({ tag }) => {
  const { colors } = useTheme()
  return (
    <View
      style={{
        borderWidth: 0.5,
        borderColor: colors.primary,
        borderRadius: 4,
        padding: 4,
        paddingTop: 5,
        marginRight: 6,
        marginTop: 8,
        height: styles.commonSize,
      }}
    >
      <RoboText color={colors.primary} uppercase weight="light" size="tiny">
        # {tag}
      </RoboText>
    </View>
  )
}

const Tags: React.FC<Props> = ({ tags }) => (
  <View
    style={{
      flexDirection: 'row',
      flexWrap: 'wrap',
    }}
  >
    {tags.map((tag, i) => (
      // eslint-disable-next-line react/no-array-index-key
      <Tag key={i} tag={tag} />
    ))}
  </View>
)

export default Tags
