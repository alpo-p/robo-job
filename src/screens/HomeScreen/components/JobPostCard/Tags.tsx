import React from 'react'

import styled from 'styled-components/native'
import RoboText from '../../../../common/components/RoboText'
import styles from '../../../../common/styles'

interface Props {
  tags: string[]
}

const TagContainer = styled.View`
  border-width: 0.5px;
  border-color: ${styles.lightGrey};
  border-radius: 10px;
  padding: 5px;
  margin: 3px;
`

const Tag: React.FC<{ tag: string }> = ({ tag }) => (
  <TagContainer>
    <RoboText weight="light" size="small">
      {tag}
    </RoboText>
  </TagContainer>
)

const Container = styled.View`
  flex-direction: row;
  flex-wrap: wrap;
`

const Tags: React.FC<Props> = ({ tags }) => {
  return (
    <Container>
      {tags.map(tag => (
        <Tag tag={tag} />
      ))}
    </Container>
  )
}

export default Tags
