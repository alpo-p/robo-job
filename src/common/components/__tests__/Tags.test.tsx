import React from 'react'
import rendererCreateWithNavigation from '../../../testUtils/rendererCreateWithNavigation'
import Tags from '../Tags'

describe('Tags', () => {
  it('should match snapshot', () => {
    const tree = rendererCreateWithNavigation(
      <Tags tags={['test1', 'test2']} />,
    )
    expect(tree).toMatchSnapshot()
  })
})
