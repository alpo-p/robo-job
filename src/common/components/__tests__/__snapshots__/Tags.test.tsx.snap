// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags should match snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "center",
        "borderColor": "#000000",
        "borderRadius": 4,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": 24,
        "marginRight": 6,
        "marginTop": 8,
        "padding": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "DamascusLight",
          "fontSize": 12,
          "textTransform": "uppercase",
        }
      }
    >
      #
       
      test1
    </Text>
  </View>
  <View
    style={
      Object {
        "alignContent": "center",
        "borderColor": "#000000",
        "borderRadius": 4,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": 24,
        "marginRight": 6,
        "marginTop": 8,
        "padding": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "DamascusLight",
          "fontSize": 12,
          "textTransform": "uppercase",
        }
      }
    >
      #
       
      test2
    </Text>
  </View>
</View>
`;
